File: .sling/libs/sushi/eqela/jk-core/0.33.0/jk.test.lua
jk = jk or {}

jk.test = jk.test or {}

jk.test.UnitTest = {}
jk.test.UnitTest.__index = jk.test.UnitTest
_vm:set_metatable(jk.test.UnitTest, {})

function jk.test.UnitTest._create()
	local v = _vm:set_metatable({}, jk.test.UnitTest)
	return v
end

function jk.test.UnitTest:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.test.UnitTest'
	self['_isType.jk.test.UnitTest'] = true
	self.ctx = _g.jk.console.ConsoleApplicationContext._construct0(_g.jk.console.ConsoleApplicationContext._create())
end

function jk.test.UnitTest:_construct0()
	jk.test.UnitTest._init(self)
	do self.ctx:setEnableDebugMessages(true) end
	return self
end

function jk.test.UnitTest:execute()
end

jk.test.FailedAssertionException = _g.jk.lang.Exception._create()
jk.test.FailedAssertionException.__index = jk.test.FailedAssertionException
_vm:set_metatable(jk.test.FailedAssertionException, {
	__index = _g.jk.lang.Exception
})

function jk.test.FailedAssertionException._create()
	local v = _vm:set_metatable({}, jk.test.FailedAssertionException)
	return v
end

function jk.test.FailedAssertionException:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.test.FailedAssertionException'
	self['_isType.jk.test.FailedAssertionException'] = true
	self['_isType.jk.lang.StringObject'] = true
	self.file = nil
	self.line = nil
	self.column = nil
	self.assertionErrorMessage = nil
end

function jk.test.FailedAssertionException:_construct0()
	jk.test.FailedAssertionException._init(self)
	do _g.jk.lang.Exception._construct0(self) end
	return self
end

function jk.test.FailedAssertionException:instance(file, line, column, message)
	local v = _g.jk.test.FailedAssertionException._construct0(_g.jk.test.FailedAssertionException._create())
	do v:setFile(file) end
	do v:setLine(line) end
	do v:setColumn(column) end
	do v:setAssertionErrorMessage(message) end
	do return v end
end

function jk.test.FailedAssertionException:toString()
	local sb = _g.jk.lang.StringBuilder._construct0(_g.jk.lang.StringBuilder._create())
	do sb:appendString(_g.jk.lang.String:safeString(self.file) .. ":" .. _g.jk.lang.String:safeString(self.line) .. ":" .. _g.jk.lang.String:safeString(self.column) .. ": " .. "FailedAssertionException") end
	if self.assertionErrorMessage ~= nil then
		do sb:appendString(": ") end
		do sb:appendString(self.assertionErrorMessage) end
	end
	do return sb:toString() end
end

function jk.test.FailedAssertionException:getFile()
	do return self.file end
end

function jk.test.FailedAssertionException:setFile(v)
	self.file = v
	do return self end
end

function jk.test.FailedAssertionException:getLine()
	do return self.line end
end

function jk.test.FailedAssertionException:setLine(v)
	self.line = v
	do return self end
end

function jk.test.FailedAssertionException:getColumn()
	do return self.column end
end

function jk.test.FailedAssertionException:setColumn(v)
	self.column = v
	do return self end
end

function jk.test.FailedAssertionException:getAssertionErrorMessage()
	do return self.assertionErrorMessage end
end

function jk.test.FailedAssertionException:setAssertionErrorMessage(v)
	self.assertionErrorMessage = v
	do return self end
end

jk.test.TestSet = {}
jk.test.TestSet.__index = jk.test.TestSet
_vm:set_metatable(jk.test.TestSet, {})

function jk.test.TestSet._create()
	local v = _vm:set_metatable({}, jk.test.TestSet)
	return v
end

function jk.test.TestSet:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.test.TestSet'
	self['_isType.jk.test.TestSet'] = true
	self.ctx = _g.jk.console.ConsoleApplicationContext._construct0(_g.jk.console.ConsoleApplicationContext._create())
	self.total = 0
	self.failed = 0
end

function jk.test.TestSet:_construct0()
	jk.test.TestSet._init(self)
	return self
end

function jk.test.TestSet:test()
end

function jk.test.TestSet:execute()
	do self:test() end
	if self.failed > 0 then
		do return 1 end
	end
	do return 0 end
end

function jk.test.TestSet:getTotal()
	do return self.total end
end

function jk.test.TestSet:setTotal(v)
	self.total = v
	do return self end
end

function jk.test.TestSet:getFailed()
	do return self.failed end
end

function jk.test.TestSet:setFailed(v)
	self.failed = v
	do return self end
end

File: .sling/libs/web5/eqela/jk-core/0.36.0/jk.test.js
let JkTestFailedAssertionException = function() {
	JkLangException.call(this);
	this.file = null;
	this.line = null;
	this.column = null;
	this.assertionErrorMessage = null;
};

JkTestFailedAssertionException.prototype = (function(o) { function F() {}; F.prototype = o; return(new F()); }) (JkLangException.prototype);
JkTestFailedAssertionException.prototype.constructor = JkTestFailedAssertionException;
JkTestFailedAssertionException.prototype._t = {
	"JkLangException" : true,
	"JkTestFailedAssertionException" : true,
	"JkLangStringObject" : true
};
JkTestFailedAssertionException.prototype._tobj = JkTestFailedAssertionException;

JkTestFailedAssertionException.NEW = function() {
	var v = new JkTestFailedAssertionException;
	return v.CTOR_JkTestFailedAssertionException();
};

JkTestFailedAssertionException.prototype.CTOR_JkTestFailedAssertionException = function() {
	this.assertionErrorMessage = null;
	this.column = null;
	this.line = null;
	this.file = null;
	if(JkLangException.prototype.CTOR_JkLangException.call(this) == null) {
		return null;
	}
	return this;
};

JkTestFailedAssertionException.instance = function(file, line, column, message) {
	var v = JkTestFailedAssertionException.NEW();
	v.setFile(file);
	v.setLine(line);
	v.setColumn(column);
	v.setAssertionErrorMessage(message);
	return v;
};

JkTestFailedAssertionException.prototype.toString = function() {
	var sb = JkLangStringBuilder.NEW();
	sb.appendString((JkLangString.safeString(this.file)) + (":") + (JkLangString.safeString(this.line)) + (":") + (JkLangString.safeString(this.column)) + (": ") + ("FailedAssertionException"));
	if(this.assertionErrorMessage != null) {
		sb.appendString(": ");
		sb.appendString(this.assertionErrorMessage);
	}
	return sb.toString();
};

JkTestFailedAssertionException.prototype.getFile = function() {
	return this.file;
};

JkTestFailedAssertionException.prototype.setFile = function(v) {
	this.file = v;
	return this;
};

JkTestFailedAssertionException.prototype.getLine = function() {
	return this.line;
};

JkTestFailedAssertionException.prototype.setLine = function(v) {
	this.line = v;
	return this;
};

JkTestFailedAssertionException.prototype.getColumn = function() {
	return this.column;
};

JkTestFailedAssertionException.prototype.setColumn = function(v) {
	this.column = v;
	return this;
};

JkTestFailedAssertionException.prototype.getAssertionErrorMessage = function() {
	return this.assertionErrorMessage;
};

JkTestFailedAssertionException.prototype.setAssertionErrorMessage = function(v) {
	this.assertionErrorMessage = v;
	return this;
};

JkTestFailedAssertionException.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkTestFailedAssertionException"] === true;
};

let JkTestUnitTest = function() {
	this.ctx = JkConsoleConsoleApplicationContext.NEW();
};

JkTestUnitTest.prototype._t = { "JkTestUnitTest" : true };
JkTestUnitTest.prototype._tobj = JkTestUnitTest;

JkTestUnitTest.NEW = function() {
	var v = new JkTestUnitTest;
	return v.CTOR_JkTestUnitTest();
};

JkTestUnitTest.prototype.CTOR_JkTestUnitTest = function() {
	this.ctx = JkConsoleConsoleApplicationContext.NEW();
	this.ctx.setEnableDebugMessages(true);
	return this;
};

JkTestUnitTest.prototype.execute = function() {};

JkTestUnitTest.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkTestUnitTest"] === true;
};

let JkTestTestSet = function() {
	this.ctx = JkConsoleConsoleApplicationContext.NEW();
	this.total = 0;
	this.failed = 0;
};

JkTestTestSet.prototype._t = { "JkTestTestSet" : true };
JkTestTestSet.prototype._tobj = JkTestTestSet;

JkTestTestSet.NEW = function() {
	var v = new JkTestTestSet;
	return v.CTOR_JkTestTestSet();
};

JkTestTestSet.prototype.CTOR_JkTestTestSet = function() {
	this.failed = 0;
	this.total = 0;
	this.ctx = JkConsoleConsoleApplicationContext.NEW();
	return this;
};

JkTestTestSet.prototype.test = function() {
};

JkTestTestSet.prototype.execute = function() {
	this.test();
	if(this.failed > 0) {
		return 1;
	}
	return 0;
};

JkTestTestSet.prototype.getTotal = function() {
	return this.total;
};

JkTestTestSet.prototype.setTotal = function(v) {
	this.total = v;
	return this;
};

JkTestTestSet.prototype.getFailed = function() {
	return this.failed;
};

JkTestTestSet.prototype.setFailed = function(v) {
	this.failed = v;
	return this;
};

JkTestTestSet.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkTestTestSet"] === true;
};
